jsk_perception/src/rect_array_actual_size_filter.cpp

```diff
@@ -143,8 +143,8 @@ namespace jsk_perception
         cv::Point3d c_3d = c_ray * (distance / c_ray.z);
         const double width = std::abs(a_3d.x - c_3d.x);
         const double height = std::abs(a_3d.y - c_3d.y);
-        if (min_x_ < width && max_x_ > width &&
-            min_y_ < height && max_y_ > height) {
+        if (min_x_ * depth.cols < width && width < max_x_ * depth.cols  &&
+            min_y_ * depth.rows < height && height < max_y_ * depth.rows) {
```

Comment:
IMO, we should add a simple test if we change the existing code in
jsk_recognition.

2016年6月2日木曜日、Kentaro Wada<notifications@github.com>さんは書きました:

> In jsk_perception/src/rect_array_actual_size_filter.cpp
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1723#discussion_r65487606>
> :
>
> > @@ -143,8 +143,8 @@ namespace jsk_perception
> >          cv::Point3d c_3d = c_ray * (distance / c_ray.z);
> >          const double width = std::abs(a_3d.x - c_3d.x);
> >          const double height = std::abs(a_3d.y - c_3d.y);
> > -        if (min_x_ < width && max_x_ > width &&
> > -            min_y_ < height && max_y_ > height) {
> > +        if (min_x_ * depth.cols < width && width < max_x_ * depth.cols  &&
> > +            min_y_ * depth.rows < height && height < max_y_ * depth.rows) {
>
> then, what is min_x_, max_x_, min_y_, and max_y_?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1723/files/31889763a750ef456b5ced3cad68a94d0d6aa66b#r65487606>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ARhe7B5GeYV4Ye-xSTYckJ53N9YMxfwXks5qHnE7gaJpZM4IsL11>
> .
>


-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com